### ISSUE TITLE
Can't setup any connection to my AdGuard instance @ home on Raspi 3

### ISSUE THREAD

**Issue by mygustavo** [JGeek00/adguard-home-manager#112]
If I want to set up a new server and click on the "Connect" button at the top right after entering all the data, I return to the main screen with the information "No saved connections" and can start again from the beginning. No error message appears. I just can't manage to set up a connection. 

I use a GrapheneOS Pixel 7 with NetGuard (all connections to AdGuard are enabled). Version of AdGuard (on Raspi 3): v0.107.43. My version of AdGuardHomeManager: 2.14.0. What could be the reason for this? Many thanks in advance.

**Comment by JGeek00** [comment]
That's weird. No one else has reported something similar. The form only closes when the connection is created. Also I don't see any error logs on my error tracker, so I don't know what could be happening.

**Comment by mygustavo** [comment]
Yes, no error message is thrown. 

On my Raspi runs Debian 11 without Docker, just and only AdGuard Home which workes as DNS for my network. 

Perhaps an additional information: I installed the GitHub Version on my Pixel 7. Then I updated this version with an version from the playstore, which wasn't successful. It didn't Start anymore. Perhaps because downloading via AuroraStore. 

So I uninstalled your app and installed the GitHub Version again. 

Since then this error occured. Before your app worked perfect. Weird. 

I would be grateful if you could provide a version with tracelogs. Because I really want to continue using your app.

Thank you so much in advance.

**Comment by JGeek00** [comment]
Try to remove data and then uninstall the app. Then install it from GitHub or Google Play. That are the official sources. Maybe that problem is because the local database is corrupted.

**Comment by mygustavo** [comment]
I did what you told me, but the result is the same 
- I search in the folder "Pixel 7/Android/data" => there is no data from you app remaining.
- I deleted Cache and Storage before uninstalling. 
- Uninstalled the app and installed the one of GitHub
- Opened it, entered my connection data and pressed the "Connect button" => no error message, just returned to the main screen with the label "No saved connections" 

Are there any other data I could delete before reinstalling?

**Comment by JGeek00** [comment]
Try this apk please
https://github.com/JGeek00/adguard-home-manager/releases/download/2.14.0_(119)/old.zip

It will send a log to my error tracker if something happens on the connection process. With that information hopefully I will know where it's failing.

**Comment by mygustavo** [comment]
Done. Hoepfully you see what's wrong :-)

**Comment by JGeek00** [comment]
Thank you. I have the log and I know on what step is failing. It's failing when it tries to fetch the server status. That step is after the login, so I really don't know why it's failing. I will send you another apk with more debugging traces.

**Comment by JGeek00** [comment]
Try this apk please
https://github.com/JGeek00/adguard-home-manager/releases/download/2.14.0_(119)/test.apk

**Comment by mygustavo** [comment]
If I want to connect I got immediately the message "Invalid username or password".

**Comment by JGeek00** [comment]
That's because the login endpoint is responding with 400, 401 or 403

Try this apk please
https://github.com/JGeek00/adguard-home-manager/releases/download/2.14.0_(119)/test2.apk

**Comment by mygustavo** [comment]
Thank you for your support. I think it's great how you help. 
So - the new version is installed, I tried to set up the connection again. The result is unfortunately still the same, but maybe you have more logging information now.

**Comment by JGeek00** [comment]
Yes, on that last apk i added more logging traces. I have found the issue. I give you some context. The serverStatus function requests data from 6 endpoints. The problem is that one of the endpoints isn't returning anything (the /stats endpoint), so that's why it's failing.

Paste this url on your browser (make sure first that you are logged in the admin panel of your server) <server_url>/control/stats, and tell me what do you receive.

Try this apk please. I added more debugging traces.
https://github.com/JGeek00/adguard-home-manager/releases/download/2.14.0_(119)/test3.apk

**Comment by mygustavo** [comment]
I would like to tell you what I receive. But this is in parts very private, because of the domains I blocked/whitelisted. 
Is where any way to send you these information via Mail?

**Comment by JGeek00** [comment]
I don't need the information of that endpoint, i only need to know if you receive some data or just an error

**Comment by mygustavo** [comment]
App "test3.apk" installed, started, tried to connect, same issue. 

Pasted URL in browser, I see a lot of data. 

Starting with "{"time_units":"days","top_queried_domains":[{"adguard.fritz.box":24629}," 
Ending with §],"num_dns_queries":540873,"num_blocked_filtering":353763,"num_replaced_safebrowsing":0,"num_replaced_safesearch":0,"num_replaced_parental":0,"avg_processing_time":0.050824}". 

In between it looks like I see all my blocked/whitelisted domains. 

But no error.

**Comment by JGeek00** [comment]
I have checked the error logs and that endpoint for whatever reason throws a timeout

**Comment by JGeek00** [comment]
Could you try to set up another AdGuard Home server on a Docker container on your raspberry or on your computer, and then try to connect with the app to check if it works?

**Comment by mygustavo** [comment]
Ok - I installed AdGuard Home as AddOn in HomeAssistant. Which credentials do I have to use for this configuration? I think it is a docker implementationen. Any idea?

**Comment by JGeek00** [comment]
Follow this guide to establish a connection to an AdGuard Home running on HA.

https://github.com/JGeek00/adguard-home-manager/wiki/Create-a-connection#home-assistant

**Comment by JGeek00** [comment]
Also, AdGuard Home running on HA it's not the same as AdGuard Home on a Docker container.

**Comment by mygustavo** [comment]
As long as I can - or you can solve my problem with this kind of installation => I'll try it :)

**Comment by mygustavo** [comment]
If I use the HomeAssistant user and password, the app tells me that user or password is wrong. 
I use the port 54321

**Comment by mygustavo** [comment]
Ok - in the meantime I have set up the AdGuard Home Server on a new Raspi - your app works great with this new installation. Looks like some setting on my old AdGuard home server was changed so that the connection with your app was no longer possible. 

Anyway - now the problem is solved. Thank you very much for your great support and I'm happy that I can now use your app again.

**Comment by JGeek00** [comment]
Great! Now you can uninstall the apk I sent you with the logging traces and install the normal app from google play or GitHub.
